Makefile.am

```diff
@@ -3082,6 +3083,32 @@ simple_access_tests_LDADD = \
     libdlopen_test_providers.la \
     $(NULL)
 
+krb5_common_test_SOURCES = \
+    src/tests/cmocka/test_krb5_common.c \
+    src/providers/krb5/krb5_child_handler.c \
+    src/providers/krb5/krb5_common.c \
+    src/providers/krb5/krb5_opts.c \
+    src/providers/data_provider_fo.c \
+    src/providers/data_provider_opts.c \
+    src/providers/data_provider_callbacks.c \
+    src/util/sss_krb5.c \
+    $(SSSD_FAILOVER_OBJ) \
+    $(NULL)
+krb5_common_test_CFLAGS = \
+    $(KRB5_CFLAGS) \
+    $(AM_CFLAGS) \
+    $(NULL)
+krb5_common_test_LDFLAGS = \
+    $(NULL)
+krb5_common_test_LDADD = \
+    $(CMOCKA_LIBS) \
+    $(KRB5_LIBS) \
+    $(CARES_LIBS) \
+    $(SSSD_LIBS) \
+    $(SSSD_INTERNAL_LTLIBS) \
+    libsss_test_common.la \
+    $(NULL)
+
 endif # HAVE_CMOCKA
```

Comment:
Some files can be removed from the test and test can be linked with existing libraries
```
@@ -3085,28 +3085,19 @@ simple_access_tests_LDADD = \
 
 krb5_common_test_SOURCES = \
     src/tests/cmocka/test_krb5_common.c \
-    src/providers/krb5/krb5_child_handler.c \
-    src/providers/krb5/krb5_common.c \
-    src/providers/krb5/krb5_opts.c \
-    src/providers/data_provider_fo.c \
-    src/providers/data_provider_opts.c \
-    src/providers/data_provider_callbacks.c \
-    src/util/sss_krb5.c \
-    $(SSSD_FAILOVER_OBJ) \
     $(NULL)
 krb5_common_test_CFLAGS = \
     $(KRB5_CFLAGS) \
     $(AM_CFLAGS) \
     $(NULL)
-krb5_common_test_LDFLAGS = \
-    $(NULL)
 krb5_common_test_LDADD = \
     $(CMOCKA_LIBS) \
-    $(KRB5_LIBS) \
-    $(CARES_LIBS) \
-    $(SSSD_LIBS) \
+    $(POPT_LIBS) \
+    $(TALLOC_LIBS) \
+    libsss_krb5_common.la \
     $(SSSD_INTERNAL_LTLIBS) \
     libsss_test_common.la \
+    libdlopen_test_providers.la \
     $(NULL)
 
 endif # HAVE_CMOCKA
```

BTW The `libsss_krb5_common` need to be before `INTERNAL_LTLIBS` otherwise there are linking errors with --as-needed